core/logs/file.go

```diff
@@ -217,8 +220,13 @@ func (w *fileLogWriter) createLogFile() (*os.File, error) {
 		return nil, err
 	}
 
+	dirperm, err := strconv.ParseInt(w.DirPerm, 8, 64)
+	if err != nil {
+		return nil, err
+	}
+
 	filepath := path.Dir(w.Filename)
-	os.MkdirAll(filepath, os.FileMode(perm))
+	os.MkdirAll(filepath, os.FileMode(dirperm))
```

Comment:
In order to keep compatibility, I think we should check `dirperm` and if it's empty, we use `perm` to create the directories

Reply:
I think compatibility has been kept 
https://github.com/beego/beego/blob/d01f2dddcc40fa8b134114143297f21f83880374/core/logs/file.go#L433
   when we register  `fileLogWriter` to `Adapter`  we have assigned a default "0770" to dirperm, 
   could you please tell any cases that user can get a  `fileLogWriter` that having a perm value meanwhile dirperm is empty
The only case I can currently think of is an explicitly configuring like:
```go
logs.SetLogger(logs.AdapterFile, `{"filename":"log/test.log", "dirperm": ""}`)
```

Reply:
Now the current users may use this function like:
```
logs.SetLogger(logs.AdapterFile, `{"filename":"log/test.log", "perm": 0770}`)
```
and we use this `perm` to create directories.

Reply:
> Now the current users may use this function like:
> 
> ```
> logs.SetLogger(logs.AdapterFile, `{"filename":"log/test.log", "perm": 0770}`)
> ```
> 
> and we use this `perm` to create directories.

that solution seems  a  compromise  and  sacrifice configurability of log file

Reply:
Yes. I think you raised a good question that we may need two `perm` fields. One is used to create directories and another is used to create log files. But now we only have one `perm` and we use it to create both directories and files, so we must avoid to introduce breaking changes.

Reply:
In fact, I think we need two `perm` fields according to above discussion. One is for creating prefix directories, which is the `dirperm` in this PR. Another is for creating and writing log files, here is `perm`. 
But, now Beego uses `perm` to create prefix directories and log files, they don't pass `dirperm`. So if we don't do this:
```go
if dirperm == ""{
    dirperm = perm
}
```
Current users may got error because they could not create the prefix directories. 

It looks not good. Or did you have any better idea?

Reply:
In the current version of this PR
if you run 
```go
logs.SetLogger(logs.AdapterFile, `{"filename":"log/test.log", "perm": 0770}`)
```
you could see that `dirperm` is absent and only `perm` is assigned, but user will not get any error and the prefix directory will be perfectly created with "0770", that beacuse when beego unmarshal user's config, DIrPerm will remain it's  default value .

only in the following case , will user get a error

``` go
logs.SetLogger(logs.AdapterFile, `{"filename":"log/test.log", "dirperm": ""}`)
```